### ISSUE TITLE
Create a job to create and update TokenList

### ISSUE THREAD

**Issue by aifury** [dataxfi/scripts#1]
### TL;DR 

We need to automate updating TokenList (list of datatokens) (https://gateway.pinata.cloud/ipfs/QmUcsbmbYT6sFTAzsoH1jtgzwi9B3RhBsZzFHjbs6igoQg) so that whenever new datatoken and datapool is created it is included in our list. Listing doesn't have to be instantaneous (at the moment of creation of data tokens) but it can be a daily/weekly job that
 
### Considerations 

1) Each network (rinkeby, ethereum, polygon, Binance smart chain, moonriver etc.) will have their own tokenlist
2) Job will be ran daily for each supported network

Below is a step-by-step details of creating tokenlist job

#### Step 1 - Fetch datatoken details

To fetch all datatokens, first fetch response from 
POST - https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query
Payload  - 
```javascript 
{
"from": 0,
"size":100,
"query": {
            "bool": { "filter":
            [
               {"terms":{"chainId":[1]}},{"term":{"_index":"aquarius"}},{"term":{"isInPurgatory":"false"}}
            ]
            }
            }
     }
```
 > You can control the pagination while looping using `**from**` and `**size**` variables in the payload above
 
 Start with `from=0` and then paginate until total number of datasets is reached. See below for the response of the above query. You can find datasets counts from the `total` variable in the response below.
 
 ```javascript 
 
 {
    "_shards": {
        "failed": 0,
        "skipped": 0,
        "successful": 10,
        "total": 10
    },
    "hits": {
        "hits": [
            {
                "_id": "did:op:12D7f75A61F3817eEaD05D26D560b7B16C4a653D",
                "_index": "aquarius",
                "_score": 0.0,
                "_source": {
                    "@context": "https://w3id.org/did/v1",
                    "accessWhiteList": [],
                    "authentication": [
                        {
                            "publicKey": "did:op:12D7f75A61F3817eEaD05D26D560b7B16C4a653D",
                            "type": "RsaSignatureAuthentication2018"
                        }
                    ],
                    "chainId": 1,
                    "created": "2020-10-28T00:48:08Z",
                    "dataToken": "0x12D7f75A61F3817eEaD05D26D560b7B16C4a653D",
                    "dataTokenInfo": {
                        "address": "0x12D7f75A61F3817eEaD05D26D560b7B16C4a653D",
                        "cap": 1000.0,
                        "decimals": 18,
                        "name": "Stunning Anemone Token",
                        "symbol": "STUANE-97"
                    },
                    
                    ......

            }
        ],
        "max_score": 0.0,
        "total": 283
    }
}
```
 
> ☝️ You will need **hits["_id"]** to get next details

Once you have **did** of dataset, send a GET request as below

GET - https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/ddo/<did:op:fcB47f5781F14Ed7E032BD395113b84C897aA23f>

Response -

```javascript

{
    "@context": "https://w3id.org/did/v1",
    "id": "did:op:fcB47f5781F14Ed7E032BD395113b84C897aA23f",
    "created": "2020-11-01T15:51:49Z",
    "updated": "2021-08-27T12:23:25Z",
    "dataToken": "0xfcB47f5781F14Ed7E032BD395113b84C897aA23f",
    "publicKey": [
        {
            "id": "did:op:fcB47f5781F14Ed7E032BD395113b84C897aA23f",
            "type": "EthereumECDSAKey",
            "owner": "0xbbd33AFa85539fa65cc08A2e61a001876D2f13FE"
        }
    ],
    "authentication": [
        {
            "type": "RsaSignatureAuthentication2018",
            "publicKey": "did:op:fcB47f5781F14Ed7E032BD395113b84C897aA23f"
        }
    ],
    "service": [],
    "proof": {},
    "price": {
        "datatoken": 446.52241599990134,
        "ocean": 20245.148626352322,
        "value": 45.48685613250311,
        "type": "pool",
        "exchange_id": "",
        "address": "0x64269Ce81a07f21cb3566aFA752810c4ca0BcB6F",
        "pools": [
            "0x64269Ce81a07f21cb3566aFA752810c4ca0BcB6F"
        ],
        "isConsumable": "true"
    },
    "isInPurgatory": "false",
    "dataTokenInfo": {
        "address": "0xfcB47f5781F14Ed7E032BD395113b84C897aA23f",
        "name": "Trenchant Pelican Token",
        "symbol": "TREPEL-36",
        "decimals": 18,
        "cap": 1000.0
    },
    "chainId": 1,
    "accessWhiteList": [],
    "event": {}
}
```
> Note that we will get all needed info about datatokens from this response above ☝️ 
First check that **price[type]** is "pool". If not, skip it.

We will need -
**1) dataTokenInfo["address"] _(this is datatoken address)_
2) dataTokenInfo["name"]
3) dataTokenInfo["symbol"]
4) dataTokenInfo["decimals"]
5) price["address"] _(this is Pool address)_**

>Note - you will need to get above info for each datatoken for each page 

#### Step 2 - Create tokenlist

Check this function to use as template to create datatoken list - https://github.com/dataxfi/datax.js/blob/main/src/TokenList.ts#L216

Using this template/function prepare datatokenlist (json file)

#### Step 3 - Publish this list in Gdrive

Now publish this list in Google drive (with open permissions to view file)

> Repeat Steps 1 to 3 for each supported network

#### Step 4 - update Configuration in Config.ts file (one time effort)

Once you publish these tokenlists (for each network)  on Gdrive, you will receive new file Ids. These file Id needs to be updated in configuration so that datax.js can fetch correct updated datatokenlist when needed (see here - https://github.com/dataxfi/datax.js/blob/main/src/Config.ts#L46 for Ethereum mainnet). 

> You might have to update function to fetchTokenList  to use Gdrive instead of IPFS (see here - https://github.com/dataxfi/datax.js/blob/main/src/TokenList.ts#L41)  

#### Step 5 -  Automate above steps (except step 4)

Once script for above steps is created, create a automated job using some scheduler like Node-Schedule (https://www.npmjs.com/package/node-schedule) to automatically execute this script everyday for all networks and update the datatokenlists in gdrive.
